Sources/Sentry/SentryANRTracker.m

```diff
@@ -6,47 +6,42 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
-#if SENTRY_HAS_UIKIT
-
 @interface
 SentryANRTracker ()
 
-@property (weak, nonatomic) id<SentryANRTrackerDelegate> delegate;
-@property (nonatomic, assign) NSTimeInterval timeoutInterval;
 @property (nonatomic, strong) id<SentryCurrentDateProvider> currentDate;
 @property (nonatomic, strong) SentryCrashWrapper *crashWrapper;
 @property (nonatomic, strong) SentryDispatchQueueWrapper *dispatchQueueWrapper;
 @property (nonatomic, strong) SentryThreadWrapper *threadWrapper;
+@property (nonatomic, strong) NSMutableSet<id<SentryANRTrackerDelegate>> *listeners;
 
 @property (weak, nonatomic) NSThread *thread;
 
 @end
 
-@implementation SentryANRTracker
+@implementation SentryANRTracker {
+    NSObject *threadLock;
+    BOOL running;
+}
 
-- (instancetype)initWithDelegate:(id<SentryANRTrackerDelegate>)delegate
-           timeoutIntervalMillis:(NSUInteger)timeoutIntervalMillis
-             currentDateProvider:(id<SentryCurrentDateProvider>)currentDateProvider
-                    crashWrapper:(SentryCrashWrapper *)crashWrapper
-            dispatchQueueWrapper:(SentryDispatchQueueWrapper *)dispatchQueueWrapper
-                   threadWrapper:(SentryThreadWrapper *)threadWrapper
+- (instancetype)initWithCurrentDateProvider:(id<SentryCurrentDateProvider>)currentDateProvider
+                               crashWrapper:(SentryCrashWrapper *)crashWrapper
+                       dispatchQueueWrapper:(SentryDispatchQueueWrapper *)dispatchQueueWrapper
+                              threadWrapper:(SentryThreadWrapper *)threadWrapper
 {
     if (self = [super init]) {
-        self.delegate = delegate;
-        self.timeoutInterval = (double)timeoutIntervalMillis / 1000;
+        self.timeoutInterval = (double)SENTRY_ANR_TRACKER_TIMEOUT_MILLIS / 1000;
```

Comment:
`h`: Shouldn't this be `options.anrTimeoutInterval`?

Reply:
Hummmm, I think it should. I even remember writing test with different timeout. I will check what happened here.

Reply:
Ow, I remember.
SentryANRTracker is accessed from the dependency container, and we don't have options there.
This is the default constructor, but the ANRIntegration changes the timeoutInterval property value.